Linux build fails with clang

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Describe the Bug

When building with `clang`, the wayland code can not be compiled. The problem occurs in multiple sunshine versions, at least from 0.14.x. When building with `-D SUNSHINE_ENABLE_WAYLAND=OFF`, it passes the compile.

### Expected Behavior

Sunshine is correctly built.

### Additional Context

The clang 14.0.6 is from the official Arch Linux repo. Ccache is enabled.

### Host Operating System

Linux

### Operating System Version

latest Arch Linux

### Architecture

64 bit

### Sunshine commit or version

0.16.0

### Package

other (self built)

### GPU Type

None

### GPU Model

not related

### GPU Driver/Mesa Version

not related

### Capture Method (Linux Only)

_No response_

### Relevant log output

```Shell
added 4 packages, and audited 5 packages in 1s

2 packages are looking for funding
run `npm fund` for details

found 0 vulnerabilities
-- The C compiler identification is Clang 14.0.6
-- The CXX compiler identification is Clang 14.0.6
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for poll
-- Looking for poll - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for gethostbyaddr_r
-- Looking for gethostbyaddr_r - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Performing Test HAS_MSGHDR_FLAGS
-- Performing Test HAS_MSGHDR_FLAGS - Success
-- Check size of socklen_t
-- Check size of socklen_t - done
-- Check size of QOS_FLOWID
-- Check size of QOS_FLOWID - failed
-- Check size of PQOS_FLOWID
-- Check size of PQOS_FLOWID - failed
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found Boost: /usr/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found suitable version "1.80.0", minimum required is "1.53.0") found components: system thread
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "3.0.7")
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0")
-- Checking for module 'libcurl'
--   Found libcurl, version 7.86.0
-- Found Boost: /usr/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0") found components: log filesystem
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - /opt/cuda/bin/nvcc
-- The CUDA compiler identification is NVIDIA 11.8.89
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /opt/cuda/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Checking for module 'libdrm'
--   Found libdrm, version 2.4.114
-- Found LIBDRM: /usr/lib/libdrm.so
-- Checking for module 'libcap'
--   Found libcap, version 2.66
-- Found LIBCAP: /usr/lib/libcap.so
-- Found Wayland: /usr/lib/libwayland-client.so;/usr/lib/libwayland-server.so;/usr/lib/libwayland-egl.so;/usr/lib/libwayland-cursor.so
wayland-scanner private-code /foo/bar/sunshine/src/sunshine/third-party/wayland-protocols/xdg-output-unstable-v1.xml /foo/bar/sunshine/src/build/generated-src/xdg-output-unstable-v1.c
wayland-scanner client-header /foo/bar/sunshine/src/sunshine/third-party/wayland-protocols/xdg-output-unstable-v1.xml /foo/bar/sunshine/src/build/generated-src/xdg-output-unstable-v1.h
wayland-scanner private-code /foo/bar/sunshine/src/sunshine/third-party/wayland-protocols/wlr-export-dmabuf-unstable-v1.xml /foo/bar/sunshine/src/build/generated-src/wlr-export-dmabuf-unstable-v1.c
wayland-scanner client-header /foo/bar/sunshine/src/sunshine/third-party/wayland-protocols/wlr-export-dmabuf-unstable-v1.xml /foo/bar/sunshine/src/build/generated-src/wlr-export-dmabuf-unstable-v1.h
-- Configuring done
-- Generating done
-- Build files have been written to: /foo/bar/sunshine/src/build
make: Entering directory '/foo/bar/sunshine/src/build'
make[1]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
[  1%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs.c.o
[  2%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_vp9.c.o
[  4%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_h2645.c.o
[  8%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_jpeg.c.o
[  8%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_sei.c.o
[ 13%] Building C object third-party/cbs/CMakeFiles/cbs.dir/h2645_parse.c.o
[ 13%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/callbacks.c.o
[ 13%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/miniupnpc.c.o
[ 14%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/minixml.c.o
[ 14%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_av1.c.o
[ 17%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/igd_desc_parse.c.o
[ 17%] Building C object third-party/cbs/CMakeFiles/cbs.dir/cbs_mpeg2.c.o
[ 18%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/minisoap.c.o
[ 20%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/minissdpc.c.o
[ 21%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/compress.c.o
[ 23%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/miniwget.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 24%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/host.c.o
[ 26%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/list.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 27%] Building C object third-party/cbs/CMakeFiles/cbs.dir/video_levels.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 30%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/upnpdev.c.o
[ 30%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/upnpcommands.c.o
[ 31%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/upnpreplyparse.c.o
[ 33%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/packet.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 34%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/peer.c.o
[ 36%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/upnperrors.c.o
[ 37%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/protocol.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 39%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/unix.c.o
[ 40%] Building C object third-party/moonlight-common-c/enet/CMakeFiles/enet.dir/win32.c.o
[ 42%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/connecthostport.c.o
[ 43%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/portlistingparse.c.o
[ 46%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/addr_is_reserved.c.o
[ 46%] Building C object third-party/miniupnp/miniupnpc/CMakeFiles/libminiupnpc-static.dir/receivedata.c.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 47%] Linking C static library libminiupnpc.a
[ 49%] Linking C static library libenet.a
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
[ 49%] Built target libminiupnpc-static
[ 49%] Built target enet
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
1 warning generated.
[ 50%] Linking C static library libcbs.a
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
[ 50%] Built target cbs
make[2]: Entering directory '/foo/bar/sunshine/src/build'
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[2]: Entering directory '/foo/bar/sunshine/src/build'
[ 53%] Building CXX object CMakeFiles/sunshine.dir/src/upnp.cpp.o
[ 53%] Building C object CMakeFiles/sunshine.dir/third-party/moonlight-common-c/reedsolomon/rs.c.o
[ 57%] Building CXX object CMakeFiles/sunshine.dir/src/stream.cpp.o
[ 59%] Building CXX object CMakeFiles/sunshine.dir/src/config.cpp.o
[ 65%] Building CXX object CMakeFiles/sunshine.dir/src/rtsp.cpp.o
[ 65%] Building CXX object CMakeFiles/sunshine.dir/src/main.cpp.o
[ 62%] Building CXX object CMakeFiles/sunshine.dir/src/confighttp.cpp.o
[ 57%] Building CXX object CMakeFiles/sunshine.dir/src/nvhttp.cpp.o
[ 63%] Building C object CMakeFiles/sunshine.dir/third-party/moonlight-common-c/src/RtspParser.c.o
[ 66%] Building CXX object CMakeFiles/sunshine.dir/src/crypto.cpp.o
[ 66%] Building CXX object CMakeFiles/sunshine.dir/src/cbs.cpp.o
[ 68%] Building CXX object CMakeFiles/sunshine.dir/src/httpcommon.cpp.o
[ 69%] Building CXX object CMakeFiles/sunshine.dir/src/video.cpp.o
[ 71%] Building CXX object CMakeFiles/sunshine.dir/src/input.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/crypto.cpp:3:
In file included from /foo/bar/sunshine/src/sunshine/src/crypto.h:12:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:320:3: warning: 'SHA256_Init' is deprecated [-Wdeprecated-declarations]
SHA256_Init(&sha256);
^
/usr/include/openssl/sha.h:73:1: note: 'SHA256_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Init(SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:321:3: warning: 'SHA256_Update' is deprecated [-Wdeprecated-declarations]
SHA256_Update(&sha256, plaintext.data(), plaintext.size());
^
/usr/include/openssl/sha.h:74:1: note: 'SHA256_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Update(SHA256_CTX *c,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:322:3: warning: 'SHA256_Final' is deprecated [-Wdeprecated-declarations]
SHA256_Final(hsh.data(), &sha256);
^
/usr/include/openssl/sha.h:76:1: note: 'SHA256_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Final(unsigned char *md, SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:417:14: warning: 'RSA_new' is deprecated [-Wdeprecated-declarations]
auto rsa = RSA_new();
^
/usr/include/openssl/rsa.h:201:1: note: 'RSA_new' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 RSA *RSA_new(void);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:418:3: warning: 'RSA_generate_key_ex' is deprecated[-Wdeprecated-declarations]
RSA_generate_key_ex(rsa, key_bits, big_num.get(), nullptr);
^
/usr/include/openssl/rsa.h:260:1: note: 'RSA_generate_key_ex' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int RSA_generate_key_ex(RSA *rsa, int bits, BIGNUM *e,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
/foo/bar/sunshine/src/sunshine/src/crypto.cpp:419:3: warning: 'EVP_PKEY_assign' is deprecated [-Wdeprecated-declarations]
EVP_PKEY_assign_RSA(pkey.get(), rsa);
^
/usr/include/openssl/evp.h:496:41: note: expanded from macro 'EVP_PKEY_assign_RSA'
#  define EVP_PKEY_assign_RSA(pkey,rsa) EVP_PKEY_assign((pkey),EVP_PKEY_RSA,\
^
/usr/include/openssl/evp.h:1327:1: note: 'EVP_PKEY_assign' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
8 warnings generated.
[ 72%] Building CXX object CMakeFiles/sunshine.dir/src/audio.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/cbs.cpp:9:
In file included from /foo/bar/sunshine/src/sunshine/src/cbs.h:4:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 73%] Building CXX object CMakeFiles/sunshine.dir/src/process.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/audio.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/common.h:14:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 75%] Building CXX object CMakeFiles/sunshine.dir/src/network.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/upnp.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.h:9:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
/foo/bar/sunshine/src/sunshine/src/upnp.cpp:20:16: warning: unused variable 'IPv6' [-Wunused-const-variable]
constexpr auto IPv6 = 1;
^
3 warnings generated.
[ 76%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/x11grab.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/network.cpp:3:
In file included from /foo/bar/sunshine/src/sunshine/src/network.h:10:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 78%] Building CUDA object CMakeFiles/sunshine.dir/src/platform/linux/cuda.cu.o
nvcc warning : The 'compute_35', 'compute_37', 'sm_35', and 'sm_37' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/input.cpp:14:
In file included from /foo/bar/sunshine/src/sunshine/src/input.h:8:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/common.h:14:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 79%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/cuda.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/config.cpp:12:
In file included from /foo/bar/sunshine/src/sunshine/src/main.h:8:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_pool.h:4:
In file included from /foo/bar/sunshine/src/sunshine/src/task_pool.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 81%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/kmsgrab.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/video.cpp:11:
In file included from /foo/bar/sunshine/src/sunshine/src/cbs.h:4:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/video.cpp:16:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:82:3: warning: 'const' qualifier on reference type 'util::it_wrap_t::reference' (aka 'typename std::iterator<std::random_access_iterator_tag, V>::value_type &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const { return *_this().get(); }
^~~~~~
/foo/bar/sunshine/src/sunshine/src/video.cpp:119:7: warning: 'set_frame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
int set_frame(AVFrame *frame) {
^
/foo/bar/sunshine/src/sunshine/src/platform/common.h:185:15: note: overridden virtual function ishere
virtual int set_frame(AVFrame *frame) {
^
In file included from /foo/bar/sunshine/src/sunshine/src/video.cpp:16:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:8:31: warning: 'iterator<std::random_access_iterator_tag, std::shared_ptr<platf::img_t>>' is deprecated [-Wdeprecated-declarations]
class it_wrap_t : public std::iterator<std::random_access_iterator_tag, V> {
^
/foo/bar/sunshine/src/sunshine/src/round_robin.h:112:30: note: in instantiation of template class'util::it_wrap_t<std::shared_ptr<platf::img_t>, util::round_robin_t<std::shared_ptr<platf::img_t>, __gnu_cxx::__normal_iterator<std::shared_ptr<platf::img_t> *, std::vector<std::shared_ptr<platf::img_t>>>>>' requested here
class round_robin_t : public it_wrap_t<V, round_robin_t<V, It>> {
^
/foo/bar/sunshine/src/sunshine/src/video.cpp:653:22: note: in instantiation of template class 'util::round_robin_t<std::shared_ptr<platf::img_t>, __gnu_cxx::__normal_iterator<std::shared_ptr<platf::img_t> *, std::vector<std::shared_ptr<platf::img_t>>>>' requested here
auto round_robin = util::make_round_robin<std::shared_ptr<platf::img_t>>(std::begin(imgs), std::end(imgs));
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_iterator_base_types.h:127:12: note: 'iterator<std::random_access_iterator_tag, std::shared_ptr<platf::img_t>>' has been explicitly marked deprecated here
struct _GLIBCXX17_DEPRECATED iterator
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/x86_64-pc-linux-gnu/bits/c++config.h:119:34: note: expanded from macro '_GLIBCXX17_DEPRECATED'
# define _GLIBCXX17_DEPRECATED [[__deprecated__]]
^
In file included from /foo/bar/sunshine/src/sunshine/src/video.cpp:16:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:11:25: warning: 'iterator<std::random_access_iterator_tag, std::shared_ptr<platf::img_t>>' is deprecated [-Wdeprecated-declarations]
typedef typename std::iterator<std::random_access_iterator_tag, V>::value_type class_t;
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_iterator_base_types.h:127:12: note: 'iterator<std::random_access_iterator_tag, std::shared_ptr<platf::img_t>>' has been explicitly marked deprecated here
struct _GLIBCXX17_DEPRECATED iterator
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/x86_64-pc-linux-gnu/bits/c++config.h:119:34: note: expanded from macro '_GLIBCXX17_DEPRECATED'
# define _GLIBCXX17_DEPRECATED [[__deprecated__]]
^
6 warnings generated.
[ 82%] Building C object CMakeFiles/sunshine.dir/generated-src/xdg-output-unstable-v1.c.o
[ 84%] Building C object CMakeFiles/sunshine.dir/generated-src/wlr-export-dmabuf-unstable-v1.c.o
[ 85%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/wlgrab.cpp.o
[ 86%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/wayland.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/rtsp.cpp:16:
In file included from /foo/bar/sunshine/src/sunshine/src/input.h:8:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/common.h:14:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
[ 88%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/publish.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/x11grab.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/common.h:14:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/x11grab.cpp:23:
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:10:1: warning: class 'hwdevice_t' was previously declared as a struct; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
class hwdevice_t;
^
/foo/bar/sunshine/src/sunshine/src/platform/common.h:174:8: note: previous use is here
struct hwdevice_t {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:10:1: note: did you mean struct here?
class hwdevice_t;
^~~~~
struct
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:11:1: warning: class 'img_t' was previously declared as a struct; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
class img_t;
^
/foo/bar/sunshine/src/sunshine/src/platform/common.h:142:8: note: previous use is here
struct img_t {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:11:1: note: did you mean struct here?
class img_t;
^~~~~
struct
4 warnings generated.
[ 89%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/vaapi.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:13:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:10:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/misc.h:7:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:13:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:12:
/foo/bar/sunshine/src/sunshine/src/platform/common.h:142:1: warning: 'img_t' defined as a struct here but previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
struct img_t {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:11:1: note: did you mean struct here?
class img_t;
^~~~~
struct
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:13:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:12:
/foo/bar/sunshine/src/sunshine/src/platform/common.h:174:1: warning: 'hwdevice_t' defined as a struct here but previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
struct hwdevice_t {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.h:10:1: note: did you mean struct here?
class hwdevice_t;
^~~~~
struct
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:183:7: warning: 'set_frame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
int set_frame(AVFrame *frame) {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:97:7: note: overridden virtual function is here
int set_frame(AVFrame *frame) override {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:29:16: warning: unused variable 'cudaDevAttrMaxThreadsPerBlock' [-Wunused-const-variable]
constexpr auto cudaDevAttrMaxThreadsPerBlock          = (CUdevice_attribute)1;
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/cuda.cpp:30:16: warning: unused variable 'cudaDevAttrMaxThreadsPerMultiProcessor' [-Wunused-const-variable]
constexpr auto cudaDevAttrMaxThreadsPerMultiProcessor = (CUdevice_attribute)39;
^
7 warnings generated.
[ 91%] Building CXX object CMakeFiles/sunshine.dir/src/platform/linux/graphics.cpp.o
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:6:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:10:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/misc.h:7:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:9:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:82:3: warning: 'const' qualifier on reference type 'util::it_wrap_t::reference' (aka 'typename std::iterator<std::random_access_iterator_tag, V>::value_type &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const { return *_this().get(); }
^~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:20:32: warning: unknown warning group '-Wpmf-conversions', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wpmf-conversions"
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:89:3: error: cannot cast from type 'void (wl::monitor_t::*)(zxdg_output_v1 *, const char *)' to pointer type 'decltype(this->listener.name)' (aka 'void (*)(void *, struct zxdg_output_v1 *, const char *)')
CLASS_CALL(listener.name, monitor_t::xdg_name);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:87:30: note: expanded from macro 'CLASS_CALL'
#define CLASS_CALL(x, y) x = (decltype(x))&y
^~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:90:3: error: cannot cast from type 'void (wl::monitor_t::*)(zxdg_output_v1 *, std::int32_t, std::int32_t)' to pointer type 'decltype(this->listener.logical_size)' (aka 'void (*)(void *, zxdg_output_v1 *, int, int)')
CLASS_CALL(listener.logical_size, monitor_t::xdg_size);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:87:30: note: expanded from macro 'CLASS_CALL'
#define CLASS_CALL(x, y) x = (decltype(x))&y
^~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:91:3: error: cannot cast from type 'void (wl::monitor_t::*)(zxdg_output_v1 *, std::int32_t, std::int32_t)' to pointer type 'decltype(this->listener.logical_position)' (aka 'void (*)(void *, zxdg_output_v1 *, int, int)')
CLASS_CALL(listener.logical_position, monitor_t::xdg_position);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:87:30: note: expanded from macro 'CLASS_CALL'
#define CLASS_CALL(x, y) x = (decltype(x))&y
^~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:92:3: error: cannot cast from type 'void (wl::monitor_t::*)(zxdg_output_v1 *)' to pointer type 'decltype(this->listener.done)' (aka 'void (*)(void *, struct zxdg_output_v1 *)')
CLASS_CALL(listener.done, monitor_t::xdg_done);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:87:30: note: expanded from macro 'CLASS_CALL'
#define CLASS_CALL(x, y) x = (decltype(x))&y
^~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:93:3: error: cannot cast from type 'void (wl::monitor_t::*)(zxdg_output_v1 *, const char *)' to pointer type 'decltype(this->listener.description)' (aka'void (*)(void *, struct zxdg_output_v1 *, const char *)')
CLASS_CALL(listener.description, monitor_t::xdg_description);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:87:30: note: expanded from macro 'CLASS_CALL'
#define CLASS_CALL(x, y) x = (decltype(x))&y
^~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:101:9: error: cannot cast from type'void (wl::interface_t::*)(wl_registry *, std::uint32_t, const char *, std::uint32_t)' to pointer type 'decltype(wl_registry_listener::global)' (aka 'void (*)(void *, wl_registry *, unsigned int, const char *, unsigned int)')
(decltype(wl_registry_listener::global))&interface_t::add_interface,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:102:9: error: cannot cast from type'void (wl::interface_t::*)(wl_registry *, uint32_t)' to pointer type 'decltype(wl_registry_listener::global_remove)'(aka 'void (*)(void *, wl_registry *, unsigned int)')
(decltype(wl_registry_listener::global_remove))&interface_t::del_interface,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:138:9: error: cannot cast from type'void (wl::dmabuf_t::*)(zwlr_export_dmabuf_frame_v1 *, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t)' to pointer type 'decltype(zwlr_export_dmabuf_frame_v1_listener::frame)' (aka 'void (*)(void *, zwlr_export_dmabuf_frame_v1 *, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)')
(decltype(zwlr_export_dmabuf_frame_v1_listener::frame))&dmabuf_t::frame,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:139:9: error: cannot cast from type'void (wl::dmabuf_t::*)(zwlr_export_dmabuf_frame_v1 *, std::uint32_t, std::int32_t, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t)' to pointer type 'decltype(zwlr_export_dmabuf_frame_v1_listener::object)' (aka 'void (*)(void *, zwlr_export_dmabuf_frame_v1 *, unsigned int, int, unsigned int, unsigned int, unsigned int, unsigned int)')
(decltype(zwlr_export_dmabuf_frame_v1_listener::object))&dmabuf_t::object,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:140:9: error: cannot cast from type'void (wl::dmabuf_t::*)(zwlr_export_dmabuf_frame_v1 *, std::uint32_t, std::uint32_t, std::uint32_t)' to pointer type'decltype(zwlr_export_dmabuf_frame_v1_listener::ready)' (aka 'void (*)(void *, zwlr_export_dmabuf_frame_v1 *, unsigned int, unsigned int, unsigned int)')
(decltype(zwlr_export_dmabuf_frame_v1_listener::ready))&dmabuf_t::ready,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/wayland.cpp:141:9: error: cannot cast from type'void (wl::dmabuf_t::*)(zwlr_export_dmabuf_frame_v1 *, zwlr_export_dmabuf_frame_v1_cancel_reason)' to pointer type 'decltype(zwlr_export_dmabuf_frame_v1_listener::cancel)' (aka 'void (*)(void *, zwlr_export_dmabuf_frame_v1 *, unsigned int)')
(decltype(zwlr_export_dmabuf_frame_v1_listener::cancel))&dmabuf_t::cancel,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
4 warnings and 11 errors generated.
make[2]: *** [CMakeFiles/sunshine.dir/build.make:413: CMakeFiles/sunshine.dir/src/platform/linux/wayland.cpp.o] Error1
make[2]: *** Waiting for unfinished jobs....
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/wlgrab.cpp:1:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/common.h:14:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_safe.h:13:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/kmsgrab.cpp:11:
In file included from /foo/bar/sunshine/src/sunshine/src/main.h:8:
In file included from /foo/bar/sunshine/src/sunshine/src/thread_pool.h:4:
In file included from /foo/bar/sunshine/src/sunshine/src/task_pool.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/kmsgrab.cpp:13:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:82:3: warning: 'const' qualifier on reference type 'util::it_wrap_t::reference' (aka 'typename std::iterator<std::random_access_iterator_tag, V>::value_type &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const { return *_this().get(); }
^~~~~~
/foo/bar/sunshine/src/sunshine/src/round_robin.h:8:31: warning: 'iterator<std::random_access_iterator_tag, _drmModePlane>' is deprecated [-Wdeprecated-declarations]
class it_wrap_t : public std::iterator<std::random_access_iterator_tag, V> {
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/kmsgrab.cpp:174:27: note: in instantiation of template class 'util::it_wrap_t<_drmModePlane, platf::kms::plane_it_t>' requested here
class plane_it_t : public util::it_wrap_t<plane_t::element_type, plane_it_t> {
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_iterator_base_types.h:127:12: note: 'iterator<std::random_access_iterator_tag, _drmModePlane>' has been explicitly marked deprecated here
struct _GLIBCXX17_DEPRECATED iterator
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/x86_64-pc-linux-gnu/bits/c++config.h:119:34: note: expanded from macro '_GLIBCXX17_DEPRECATED'
# define _GLIBCXX17_DEPRECATED [[__deprecated__]]
^
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/kmsgrab.cpp:13:
/foo/bar/sunshine/src/sunshine/src/round_robin.h:11:25: warning: 'iterator<std::random_access_iterator_tag, _drmModePlane>' is deprecated [-Wdeprecated-declarations]
typedef typename std::iterator<std::random_access_iterator_tag, V>::value_type class_t;
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_iterator_base_types.h:127:12: note: 'iterator<std::random_access_iterator_tag, _drmModePlane>' has been explicitly marked deprecated here
struct _GLIBCXX17_DEPRECATED iterator
^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/x86_64-pc-linux-gnu/bits/c++config.h:119:34: note: expanded from macro '_GLIBCXX17_DEPRECATED'
# define _GLIBCXX17_DEPRECATED [[__deprecated__]]
^
/foo/bar/sunshine/src/sunshine/src/platform/linux/kmsgrab.cpp:780:13: warning: 'capture' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
capture_e capture(snapshot_cb_t &&snapshot_cb, std::shared_ptr<img_t> img, bool *cursor) {
^
/foo/bar/sunshine/src/sunshine/src/platform/common.h:228:21: note: overridden virtual function ishere
virtual capture_e capture(snapshot_cb_t &&snapshot_cb, std::shared_ptr<img_t> img, bool *cursor) = 0;
^
6 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/publish.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/misc.h:7:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/httpcommon.cpp:3:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/main.cpp:3:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/stream.cpp:3:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/vaapi.cpp:10:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:10:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/misc.h:7:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.cpp:1:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.h:10:
In file included from /foo/bar/sunshine/src/sunshine/src/platform/linux/misc.h:7:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.cpp:34:6: warning: logical not is onlyapplied to the left hand side of this comparison [-Wlogical-not-parentheses]
if(!size() == 0) {
^       ~~
/foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.cpp:34:6: note: add parentheses after the '!' to evaluate the comparison first
if(!size() == 0) {
^
(          )
/foo/bar/sunshine/src/sunshine/src/platform/linux/graphics.cpp:34:6: note: add parentheses aroundleft hand side expression to silence this warning
if(!size() == 0) {
^
(      )
3 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/process.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:6:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:104:7: warning: 'MD5' is deprecated [-Wdeprecated-declarations]
MD5(reinterpret_cast<const unsigned char *>(&input[0]), input.size(), reinterpret_cast<unsigned char *>(&hash[0]));
^
/usr/include/openssl/md5.h:52:1: note: 'MD5' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:107:9: warning: 'MD5' is deprecated [-Wdeprecated-declarations]
MD5(reinterpret_cast<const unsigned char *>(&hash[0]), hash.size(), reinterpret_cast<unsigned char *>(&hash[0]));
^
/usr/include/openssl/md5.h:52:1: note: 'MD5' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:114:7: warning: 'MD5_Init' is deprecated [-Wdeprecated-declarations]
MD5_Init(&context);
^
/usr/include/openssl/md5.h:49:1: note: 'MD5_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Init(MD5_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:118:9: warning: 'MD5_Update' is deprecated [-Wdeprecated-declarations]
MD5_Update(&context, buffer.data(), static_cast<std::size_t>(read_length));
^
/usr/include/openssl/md5.h:50:1: note: 'MD5_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Update(MD5_CTX *c, const void *data, size_t len);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:121:7: warning: 'MD5_Final' is deprecated [-Wdeprecated-declarations]
MD5_Final(reinterpret_cast<unsigned char *>(&hash[0]), &context);
^
/usr/include/openssl/md5.h:51:1: note: 'MD5_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Final(unsigned char *md, MD5_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:124:9: warning: 'MD5' is deprecated [-Wdeprecated-declarations]
MD5(reinterpret_cast<const unsigned char *>(&hash[0]), hash.size(), reinterpret_cast<unsigned char *>(&hash[0]));
^
/usr/include/openssl/md5.h:52:1: note: 'MD5' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:143:7: warning: 'SHA1_Init' is deprecated [-Wdeprecated-declarations]
SHA1_Init(&context);
^
/usr/include/openssl/sha.h:49:1: note: 'SHA1_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA1_Init(SHA_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:147:9: warning: 'SHA1_Update' is deprecated [-Wdeprecated-declarations]
SHA1_Update(&context, buffer.data(), static_cast<std::size_t>(read_length));
^
/usr/include/openssl/sha.h:50:1: note: 'SHA1_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA1_Update(SHA_CTX *c, const void *data, size_t len);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:150:7: warning: 'SHA1_Final' is deprecated [-Wdeprecated-declarations]
SHA1_Final(reinterpret_cast<unsigned char *>(&hash[0]), &context);
^
/usr/include/openssl/sha.h:51:1: note: 'SHA1_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA1_Final(unsigned char *md, SHA_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:172:7: warning: 'SHA256_Init' is deprecated [-Wdeprecated-declarations]
SHA256_Init(&context);
^
/usr/include/openssl/sha.h:73:1: note: 'SHA256_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Init(SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:176:9: warning: 'SHA256_Update' is deprecated [-Wdeprecated-declarations]
SHA256_Update(&context, buffer.data(), static_cast<std::size_t>(read_length));
^
/usr/include/openssl/sha.h:74:1: note: 'SHA256_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Update(SHA256_CTX *c,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:179:7: warning: 'SHA256_Final' is deprecated [-Wdeprecated-declarations]
SHA256_Final(reinterpret_cast<unsigned char *>(&hash[0]), &context);
^
/usr/include/openssl/sha.h:76:1: note: 'SHA256_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Final(unsigned char *md, SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:201:7: warning: 'SHA512_Init' is deprecated [-Wdeprecated-declarations]
SHA512_Init(&context);
^
/usr/include/openssl/sha.h:123:1: note: 'SHA512_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA512_Init(SHA512_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:205:9: warning: 'SHA512_Update' is deprecated [-Wdeprecated-declarations]
SHA512_Update(&context, buffer.data(), static_cast<std::size_t>(read_length));
^
/usr/include/openssl/sha.h:124:1: note: 'SHA512_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA512_Update(SHA512_CTX *c,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
In file included from /foo/bar/sunshine/src/sunshine/src/confighttp.cpp:20:
/foo/bar/sunshine/src/sunshine/third-party/Simple-Web-Server/crypto.hpp:208:7: warning: 'SHA512_Final' is deprecated [-Wdeprecated-declarations]
SHA512_Final(reinterpret_cast<unsigned char *>(&hash[0]), &context);
^
/usr/include/openssl/sha.h:126:1: note: 'SHA512_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA512_Final(unsigned char *md, SHA512_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
^
17 warnings generated.
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from /foo/bar/sunshine/src/sunshine/src/nvhttp.cpp:5:
In file included from /foo/bar/sunshine/src/sunshine/src/process.h:15:
/foo/bar/sunshine/src/sunshine/src/utility.h:667:3: warning: 'const' qualifier on reference type 'util::wrap_ptr::reference' (aka 'T &') has no effect [-Wignored-reference-qualifiers]
const reference operator*() const {
^~~~~~
2 warnings generated.
make[2]: Leaving directory '/foo/bar/sunshine/src/build'
make[1]: *** [CMakeFiles/Makefile2:156: CMakeFiles/sunshine.dir/all] Error 2
make[1]: Leaving directory '/foo/bar/sunshine/src/build'
make: *** [Makefile:156: all] Error 2
make: Leaving directory '/foo/bar/sunshine/src/build'
==> ERROR: A failure occurred in build().
Aborting...
```

--- a/src/platform/linux/wayland.cpp
+++ b/src/platform/linux/wayland.cpp
@@ -22,12 +22,12 @@ using namespace std::literals;
 namespace wl {
 
 // Helper to call C++ method from wayland C callback
-template<class T, class Method, Method m, class ...Params>
+template<class T, class Method, Method m, class... Params>
 static auto classCall(void *data, Params... params) -> decltype(((*reinterpret_cast<T *>(data)).*m)(params...)) {
   return ((*reinterpret_cast<T *>(data)).*m)(params...);
 }
 
-#define CLASS_CALL(c,m) classCall<c, decltype(&c::m), &c::m>
+#define CLASS_CALL(c, m) classCall<c, decltype(&c::m), &c::m>
 
 int display_t::init(const char *display_name) {
   if(!display_name) {
@@ -60,12 +60,12 @@ wl_registry *display_t::registry() {
 
 inline monitor_t::monitor_t(wl_output *output)
     : output { output }, listener {
-      &CLASS_CALL(monitor_t, xdg_position),
-      &CLASS_CALL(monitor_t, xdg_size),
-      &CLASS_CALL(monitor_t, xdg_done),
-      &CLASS_CALL(monitor_t, xdg_name),
-      &CLASS_CALL(monitor_t, xdg_description)
-    } {}
+        &CLASS_CALL(monitor_t, xdg_position),
+        &CLASS_CALL(monitor_t, xdg_size),
+        &CLASS_CALL(monitor_t, xdg_done),
+        &CLASS_CALL(monitor_t, xdg_name),
+        &CLASS_CALL(monitor_t, xdg_description)
+      } {}
 
 inline void monitor_t::xdg_name(zxdg_output_v1 *, const char *name) {
   this->name = name;

--- a/src/platform/linux/wayland.cpp
+++ b/src/platform/linux/wayland.cpp
@@ -20,6 +20,15 @@ using namespace std::literals;
 #pragma GCC diagnostic ignored "-Wpmf-conversions"
 
 namespace wl {
+
+// Helper to call C++ method from wayland C callback
+template<class T, class Method, Method m, class ...Params>
+static auto classCall(void *data, Params... params) -> decltype(((*reinterpret_cast<T *>(data)).*m)(params...)) {
+  return ((*reinterpret_cast<T *>(data)).*m)(params...);
+}
+
+#define CLASS_CALL(c,m) classCall<c, decltype(&c::m), &c::m>
+
 int display_t::init(const char *display_name) {
   if(!display_name) {
     display_name = std::getenv("WAYLAND_DISPLAY");
@@ -49,7 +58,14 @@ wl_registry *display_t::registry() {
   return wl_display_get_registry(display_internal.get());
 }
 
-inline monitor_t::monitor_t(wl_output *output) : output { output } {}
+inline monitor_t::monitor_t(wl_output *output)
+    : output { output }, listener {
+      &CLASS_CALL(monitor_t, xdg_position),
+      &CLASS_CALL(monitor_t, xdg_size),
+      &CLASS_CALL(monitor_t, xdg_done),
+      &CLASS_CALL(monitor_t, xdg_name),
+      &CLASS_CALL(monitor_t, xdg_description)
+    } {}
 
 inline void monitor_t::xdg_name(zxdg_output_v1 *, const char *name) {
   this->name = name;
@@ -83,23 +99,13 @@ void monitor_t::xdg_done(zxdg_output_v1 *) {
 
 void monitor_t::listen(zxdg_output_manager_v1 *output_manager) {
   auto xdg_output = zxdg_output_manager_v1_get_xdg_output(output_manager, output);
-
-#define CLASS_CALL(x, y) x = (decltype(x))&y
-
-  CLASS_CALL(listener.name, monitor_t::xdg_name);
-  CLASS_CALL(listener.logical_size, monitor_t::xdg_size);
-  CLASS_CALL(listener.logical_position, monitor_t::xdg_position);
-  CLASS_CALL(listener.done, monitor_t::xdg_done);
-  CLASS_CALL(listener.description, monitor_t::xdg_description);
-
-#undef CLASS_CALL
   zxdg_output_v1_add_listener(xdg_output, &listener, this);
 }
 
 interface_t::interface_t() noexcept
     : output_manager { nullptr }, listener {
-        (decltype(wl_registry_listener::global))&interface_t::add_interface,
-        (decltype(wl_registry_listener::global_remove))&interface_t::del_interface,
+        &CLASS_CALL(interface_t, add_interface),
+        &CLASS_CALL(interface_t, del_interface)
       } {}
 
 void interface_t::listen(wl_registry *registry) {
@@ -135,10 +141,10 @@ void interface_t::del_interface(wl_registry *registry, uint32_t id) {
 
 dmabuf_t::dmabuf_t()
     : status { READY }, frames {}, current_frame { &frames[0] }, listener {
-        (decltype(zwlr_export_dmabuf_frame_v1_listener::frame))&dmabuf_t::frame,
-        (decltype(zwlr_export_dmabuf_frame_v1_listener::object))&dmabuf_t::object,
-        (decltype(zwlr_export_dmabuf_frame_v1_listener::ready))&dmabuf_t::ready,
-        (decltype(zwlr_export_dmabuf_frame_v1_listener::cancel))&dmabuf_t::cancel,
+        &CLASS_CALL(dmabuf_t, frame),
+        &CLASS_CALL(dmabuf_t, object),
+        &CLASS_CALL(dmabuf_t, ready),
+        &CLASS_CALL(dmabuf_t, cancel)
       } {
 }
 
@@ -200,7 +206,7 @@ void dmabuf_t::ready(
 
 void dmabuf_t::cancel(
   zwlr_export_dmabuf_frame_v1 *frame,
-  zwlr_export_dmabuf_frame_v1_cancel_reason reason) {
+  std::uint32_t reason) {
 
   zwlr_export_dmabuf_frame_v1_destroy(frame);
 

--- a/src/platform/linux/wayland.h
+++ b/src/platform/linux/wayland.h
@@ -71,7 +71,7 @@ public:
 
   void cancel(
     zwlr_export_dmabuf_frame_v1 *frame,
-    zwlr_export_dmabuf_frame_v1_cancel_reason reason);
+    std::uint32_t reason);
 
   inline frame_t *get_next_frame() {
     return current_frame == &frames[0] ? &frames[1] : &frames[0];

